shuup/discounts/modules.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2018, Shuup Inc. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from django.utils.translation import ugettext_lazy as _
+
+from shuup.core.models import ShopProduct
+from shuup.core.order_creator import OrderSourceModifierModule
+from shuup.core.pricing import DiscountModule
+
+from .models import CouponCode, CouponUsage
+from .utils import (
+    get_active_discount_for_code, get_potential_discounts_for_product
+)
+
+
+class ProductDiscountModule(DiscountModule):
+    identifier = "product_discounts"
+    name = _("Product Discounts")
+
+    def discount_price(self, context, product, price_info):
+        shop = context.shop
```

Comment:
when `shops` is `None` the campaign is not being applied

Reply:
That is exactly what we want atm.